Core/CwCheat.cpp

```diff
@@ -474,6 +477,8 @@ CheatOperation CWCheatEngine::InterpretNextCwCheat(const CheatCode &cheat, size_
 
 	// Filled as needed.
 	u32 addr;
+	leftVibration = 0;
+	rightVibration = 0;
```

Comment:
You don't want to set this to 0 here - this might be the problem.  This function is called per each cheat.

The design I'd recommend here is:

1. I'd suggest we have sceCtrl "own" the vibration state.  I don't think CwCheat should own it.
2. When CwCheat interprets or perhaps executes this, it calls a function telling sceCtrl what the new vibration should be (never setting it to zero.)
3. When sceCtrl's vblank handler happens, reduce the vibration states by a set %.  This will make vibrations back off unless the cheat keeps reactivating.
4. Have xinput read the vibration state from sceCtrl basically as you already have it.

The important thing is we never want to reset it to zero (even if we moved this outside of each cheat op): xinput runs on a separate thread, and reads this value at "random" times.  Even if we always set it back to non-zero, there's a window of time where it's zero - so it'll always potentially be buggy.

-[Unknown]